docs/design/generics/overview.md

```diff
@@ -43,51 +43,79 @@ pointers to other design documents that dive deeper into individual topics.
 
 ## Goals
 
-The goal of Carbon generics is to provide an alternative to Carbon (or C++)
-templates. Generics in this form should provide many advantages, including:
+Carbon [generics](terminology.md#generic-means-compile-time-parameterized)
+supports generalizing code to apply to more situations by adding compile-time
+parameters, allowing
+[generic programming](https://en.wikipedia.org/wiki/Generic_programming). Carbon
+supports both
+[checked and template](terminology.md#checked-versus-template-parameters)
+generics.
+
+Template generics support a similar model as C++ templates, to help with interop
+and migration. They can be more convenient to write, and support some use cases,
+like [metaprogramming](https://en.wikipedia.org/wiki/Metaprogramming), that are
+difficult with checked generics.
```

Comment:
```suggestion
Template generics provide a similar model to C++ templates, to help with interop
and migration. They can be more convenient to write, and support some use cases,
like [metaprogramming](https://en.wikipedia.org/wiki/Metaprogramming), that are
difficult with checked generics.
```

I think a word other than "support" works better here in the context of the previous paragraph (I found the repeated use of "support" with different meanings hard to read).